### PR TITLE
[quantization] Enabling inplace relu

### PR DESCRIPTION
**Pull request by z-a-f**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#28710 [quantization] Enabling inplace relu**

Differential Revision: [D18146120](https://our.internmc.facebook.com/intern/diff/D18146120)

### CODE REVIEW

**Comment by jerryzh168** [comment]
does this work? why do we put this in the beginning?

**Review by jerryzh168 — APPROVED** [review]

**Comment by z-a-f** [comment]
> does this work? why do we put this assert in the beginning?

Because we didn't support inplace initially, and there was a similar assertion in the functional.relu. After enabling the inplace, the assertion was never lifted in the module (only in the functional)

File: torch/nn/quantized/modules/activation.py [hunk 1]
```diff
@@ -31,7 +31,6 @@ class ReLU(torch.nn.ReLU):
     """
     def __init__(self, inplace=False):
         super(ReLU, self).__init__(inplace)
-        assert not inplace, 'torch.nn.quantized.ReLU does not support inplace'
 
     def forward(self, input):
         return torch.nn.quantized.functional.relu(input)
```
Comment (vadimkantorov) [inline torch/nn/quantized/modules/activation.py]: should `inplace = self.inplace` be then passed to `torch.nn.quantized.functional.relu` call?

Comment (z-a-f) [reply torch/nn/quantized/modules/activation.py]: Good point! Missed that!

**Pull request closed without merging**